### PR TITLE
Improve Compatibility Drawer in Product Page: Compatibility Link Is Used More Reasonably 

### PR DESCRIPTION
**Pull request by danielcliu-ifixit**
## Overview 

We weren't rendering the compatibility link when we had devices with truncated models lists, leading to a confusing UI. Now if we have to truncate the model lists, we will always show the compatibility links so that people will be able to click and see all of the models.

## CR

Is there a better way to make this check?

## QA

Make sure that a device with more than 3 variants has the link, that a device with more than 4 models has the link, that a device with both has the link, and a device with neither doesn't have the link. 

Closes: #931

### CODE REVIEW

**Comment by deltuh-vee** [comment]
QA 🎬 
- [x] [Devices](https://react-commerce-git-product-page-v2-compatability-c4aee2-ifixit.vercel.app/products/macbook-pro-14-2021-a2442-and-16-2021-a2485-heat-sink-brackets) with more than 3 variants has the compatibility link
- [x] [Devices](https://react-commerce-git-product-page-v2-compatability-c4aee2-ifixit.vercel.app/products/garmin-nuvi-2539-2559-2589-2597-2599-replacement-battery) with more than 4 models has the compatibility link
- [x] [Devices](https://react-commerce-git-product-page-v2-compatability-c4aee2-ifixit.vercel.app/products/hp-pavilion-g4-g6-and-g7-fan#compatibility) with both 3 variants and 4 models has the compatibility link
- [x] [Devices](https://react-commerce-git-product-page-v2-compatability-c4aee2-ifixit.vercel.app/products/airpods-max-headphone-earpad-cushions) with neither doesn't have the compatibility link

File: frontend/templates/product/sections/ProductSection/index.tsx [hunk 1]
```diff
@@ -73,6 +73,19 @@ export function ProductSection({
...
 
+   const compatibilityDrawerModelsTruncate = 4;
+   const compatibilityDrawerDeviceTruncate = 3;
+   const compatibilityDrawerIncomplete =
+      product.compatibility &&
+      (product.compatibility.devices.length >
+         compatibilityDrawerDeviceTruncate ||
+         !product.compatibility.devices
+            .slice(0, compatibilityDrawerDeviceTruncate)
+            .every(
```
Comment (danielbeardsley) [inline frontend/templates/product/sections/ProductSection/index.tsx:84]: Why `.every()` and the reverse boolean over `.some(... length > ModelsTruncate)` ? Seems harder to read.

Comment (danielcliu-ifixit) [reply frontend/templates/product/sections/ProductSection/index.tsx:84]: uh i missed that function when looking over the docs. Yeah thats much better

File: frontend/templates/product/sections/ProductSection/index.tsx [hunk 2]
```diff
@@ -73,6 +73,19 @@ export function ProductSection({
...
    const isForSale = useIsProductForSale(product);
 
+   const compatibilityDrawerModelsTruncate = 4;
+   const compatibilityDrawerDeviceTruncate = 3;
+   const compatibilityDrawerIncomplete =
+      product.compatibility &&
+      (product.compatibility.devices.length >
+         compatibilityDrawerDeviceTruncate ||
+         product.compatibility.devices
+            .slice(0, compatibilityDrawerDeviceTruncate)
```
Comment (danielbeardsley) [inline frontend/templates/product/sections/ProductSection/index.tsx:83]:
```suggestion
```

This isn't needed cause if it was longer than `Truncate`, the first part of the `||` would be true and this wouldn't be evaluated.

**Comment by danielbeardsley** [comment]
CR :+1:

File: frontend/templates/product/sections/ProductSection/index.tsx (hunk 2 above)
Comment (danielcliu-ifixit) [reply frontend/templates/product/sections/ProductSection/index.tsx]: good point

**Review by danielbeardsley — APPROVED** [review]
CR :+1:

**Comment by deltuh-vee** [comment]
Carry Over QA 🎬

**Pull request merged**